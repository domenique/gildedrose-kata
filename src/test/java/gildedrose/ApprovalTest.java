package gildedrose;

import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.Test;


class ApprovalTest {

    @Test
    void approvalTest() {
        String[] names = {"foo", "Aged Brie", "Backstage passes to a TAFKAL80ETC concert", "Sulfuras, Hand of Ragnaros", "Conjured Mana Cake"};
        Integer[] sellIns = {0, 1, 2, 3, 4, 10};
        Integer[] qualities = {0, 1, 2, 3, 10, 20};
        CombinationApprovals.verifyAllCombinations(this::runIt, names, sellIns, qualities);
    }

    private String runIt(String name, int sellIn, int quality) {
        var items = new Item[]{new Item(name, sellIn, quality)};
        GildedRose app = new GildedRose(items);
        app.updateQuality();
        return app.items[0].toString();
    }
}
