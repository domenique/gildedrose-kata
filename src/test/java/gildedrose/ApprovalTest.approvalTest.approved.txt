[foo, 0, 0] => foo,-1,0 
[foo, 0, 1] => foo,-1,0 
[foo, 0, 2] => foo,-1,0 
[foo, 0, 3] => foo,-1,1 
[foo, 0, 10] => foo,-1,8 
[foo, 0, 20] => foo,-1,18 
[foo, 1, 0] => foo,0,0 
[foo, 1, 1] => foo,0,0 
[foo, 1, 2] => foo,0,1 
[foo, 1, 3] => foo,0,2 
[foo, 1, 10] => foo,0,9 
[foo, 1, 20] => foo,0,19 
[foo, 2, 0] => foo,1,0 
[foo, 2, 1] => foo,1,0 
[foo, 2, 2] => foo,1,1 
[foo, 2, 3] => foo,1,2 
[foo, 2, 10] => foo,1,9 
[foo, 2, 20] => foo,1,19 
[foo, 3, 0] => foo,2,0 
[foo, 3, 1] => foo,2,0 
[foo, 3, 2] => foo,2,1 
[foo, 3, 3] => foo,2,2 
[foo, 3, 10] => foo,2,9 
[foo, 3, 20] => foo,2,19 
[foo, 4, 0] => foo,3,0 
[foo, 4, 1] => foo,3,0 
[foo, 4, 2] => foo,3,1 
[foo, 4, 3] => foo,3,2 
[foo, 4, 10] => foo,3,9 
[foo, 4, 20] => foo,3,19 
[foo, 10, 0] => foo,9,0 
[foo, 10, 1] => foo,9,0 
[foo, 10, 2] => foo,9,1 
[foo, 10, 3] => foo,9,2 
[foo, 10, 10] => foo,9,9 
[foo, 10, 20] => foo,9,19 
[Aged Brie, 0, 0] => Aged Brie,-1,2 
[Aged Brie, 0, 1] => Aged Brie,-1,3 
[Aged Brie, 0, 2] => Aged Brie,-1,4 
[Aged Brie, 0, 3] => Aged Brie,-1,5 
[Aged Brie, 0, 10] => Aged Brie,-1,12 
[Aged Brie, 0, 20] => Aged Brie,-1,22 
[Aged Brie, 1, 0] => Aged Brie,0,1 
[Aged Brie, 1, 1] => Aged Brie,0,2 
[Aged Brie, 1, 2] => Aged Brie,0,3 
[Aged Brie, 1, 3] => Aged Brie,0,4 
[Aged Brie, 1, 10] => Aged Brie,0,11 
[Aged Brie, 1, 20] => Aged Brie,0,21 
[Aged Brie, 2, 0] => Aged Brie,1,1 
[Aged Brie, 2, 1] => Aged Brie,1,2 
[Aged Brie, 2, 2] => Aged Brie,1,3 
[Aged Brie, 2, 3] => Aged Brie,1,4 
[Aged Brie, 2, 10] => Aged Brie,1,11 
[Aged Brie, 2, 20] => Aged Brie,1,21 
[Aged Brie, 3, 0] => Aged Brie,2,1 
[Aged Brie, 3, 1] => Aged Brie,2,2 
[Aged Brie, 3, 2] => Aged Brie,2,3 
[Aged Brie, 3, 3] => Aged Brie,2,4 
[Aged Brie, 3, 10] => Aged Brie,2,11 
[Aged Brie, 3, 20] => Aged Brie,2,21 
[Aged Brie, 4, 0] => Aged Brie,3,1 
[Aged Brie, 4, 1] => Aged Brie,3,2 
[Aged Brie, 4, 2] => Aged Brie,3,3 
[Aged Brie, 4, 3] => Aged Brie,3,4 
[Aged Brie, 4, 10] => Aged Brie,3,11 
[Aged Brie, 4, 20] => Aged Brie,3,21 
[Aged Brie, 10, 0] => Aged Brie,9,1 
[Aged Brie, 10, 1] => Aged Brie,9,2 
[Aged Brie, 10, 2] => Aged Brie,9,3 
[Aged Brie, 10, 3] => Aged Brie,9,4 
[Aged Brie, 10, 10] => Aged Brie,9,11 
[Aged Brie, 10, 20] => Aged Brie,9,21 
[Backstage passes to a TAFKAL80ETC concert, 0, 0] => Backstage passes to a TAFKAL80ETC concert,-1,0 
[Backstage passes to a TAFKAL80ETC concert, 0, 1] => Backstage passes to a TAFKAL80ETC concert,-1,0 
[Backstage passes to a TAFKAL80ETC concert, 0, 2] => Backstage passes to a TAFKAL80ETC concert,-1,0 
[Backstage passes to a TAFKAL80ETC concert, 0, 3] => Backstage passes to a TAFKAL80ETC concert,-1,0 
[Backstage passes to a TAFKAL80ETC concert, 0, 10] => Backstage passes to a TAFKAL80ETC concert,-1,0 
[Backstage passes to a TAFKAL80ETC concert, 0, 20] => Backstage passes to a TAFKAL80ETC concert,-1,0 
[Backstage passes to a TAFKAL80ETC concert, 1, 0] => Backstage passes to a TAFKAL80ETC concert,0,3 
[Backstage passes to a TAFKAL80ETC concert, 1, 1] => Backstage passes to a TAFKAL80ETC concert,0,4 
[Backstage passes to a TAFKAL80ETC concert, 1, 2] => Backstage passes to a TAFKAL80ETC concert,0,5 
[Backstage passes to a TAFKAL80ETC concert, 1, 3] => Backstage passes to a TAFKAL80ETC concert,0,6 
[Backstage passes to a TAFKAL80ETC concert, 1, 10] => Backstage passes to a TAFKAL80ETC concert,0,13 
[Backstage passes to a TAFKAL80ETC concert, 1, 20] => Backstage passes to a TAFKAL80ETC concert,0,23 
[Backstage passes to a TAFKAL80ETC concert, 2, 0] => Backstage passes to a TAFKAL80ETC concert,1,3 
[Backstage passes to a TAFKAL80ETC concert, 2, 1] => Backstage passes to a TAFKAL80ETC concert,1,4 
[Backstage passes to a TAFKAL80ETC concert, 2, 2] => Backstage passes to a TAFKAL80ETC concert,1,5 
[Backstage passes to a TAFKAL80ETC concert, 2, 3] => Backstage passes to a TAFKAL80ETC concert,1,6 
[Backstage passes to a TAFKAL80ETC concert, 2, 10] => Backstage passes to a TAFKAL80ETC concert,1,13 
[Backstage passes to a TAFKAL80ETC concert, 2, 20] => Backstage passes to a TAFKAL80ETC concert,1,23 
[Backstage passes to a TAFKAL80ETC concert, 3, 0] => Backstage passes to a TAFKAL80ETC concert,2,3 
[Backstage passes to a TAFKAL80ETC concert, 3, 1] => Backstage passes to a TAFKAL80ETC concert,2,4 
[Backstage passes to a TAFKAL80ETC concert, 3, 2] => Backstage passes to a TAFKAL80ETC concert,2,5 
[Backstage passes to a TAFKAL80ETC concert, 3, 3] => Backstage passes to a TAFKAL80ETC concert,2,6 
[Backstage passes to a TAFKAL80ETC concert, 3, 10] => Backstage passes to a TAFKAL80ETC concert,2,13 
[Backstage passes to a TAFKAL80ETC concert, 3, 20] => Backstage passes to a TAFKAL80ETC concert,2,23 
[Backstage passes to a TAFKAL80ETC concert, 4, 0] => Backstage passes to a TAFKAL80ETC concert,3,3 
[Backstage passes to a TAFKAL80ETC concert, 4, 1] => Backstage passes to a TAFKAL80ETC concert,3,4 
[Backstage passes to a TAFKAL80ETC concert, 4, 2] => Backstage passes to a TAFKAL80ETC concert,3,5 
[Backstage passes to a TAFKAL80ETC concert, 4, 3] => Backstage passes to a TAFKAL80ETC concert,3,6 
[Backstage passes to a TAFKAL80ETC concert, 4, 10] => Backstage passes to a TAFKAL80ETC concert,3,13 
[Backstage passes to a TAFKAL80ETC concert, 4, 20] => Backstage passes to a TAFKAL80ETC concert,3,23 
[Backstage passes to a TAFKAL80ETC concert, 10, 0] => Backstage passes to a TAFKAL80ETC concert,9,2 
[Backstage passes to a TAFKAL80ETC concert, 10, 1] => Backstage passes to a TAFKAL80ETC concert,9,3 
[Backstage passes to a TAFKAL80ETC concert, 10, 2] => Backstage passes to a TAFKAL80ETC concert,9,4 
[Backstage passes to a TAFKAL80ETC concert, 10, 3] => Backstage passes to a TAFKAL80ETC concert,9,5 
[Backstage passes to a TAFKAL80ETC concert, 10, 10] => Backstage passes to a TAFKAL80ETC concert,9,12 
[Backstage passes to a TAFKAL80ETC concert, 10, 20] => Backstage passes to a TAFKAL80ETC concert,9,22 
[Sulfuras, Hand of Ragnaros, 0, 0] => Sulfuras, Hand of Ragnaros,0,0 
[Sulfuras, Hand of Ragnaros, 0, 1] => Sulfuras, Hand of Ragnaros,0,1 
[Sulfuras, Hand of Ragnaros, 0, 2] => Sulfuras, Hand of Ragnaros,0,2 
[Sulfuras, Hand of Ragnaros, 0, 3] => Sulfuras, Hand of Ragnaros,0,3 
[Sulfuras, Hand of Ragnaros, 0, 10] => Sulfuras, Hand of Ragnaros,0,10 
[Sulfuras, Hand of Ragnaros, 0, 20] => Sulfuras, Hand of Ragnaros,0,20 
[Sulfuras, Hand of Ragnaros, 1, 0] => Sulfuras, Hand of Ragnaros,1,0 
[Sulfuras, Hand of Ragnaros, 1, 1] => Sulfuras, Hand of Ragnaros,1,1 
[Sulfuras, Hand of Ragnaros, 1, 2] => Sulfuras, Hand of Ragnaros,1,2 
[Sulfuras, Hand of Ragnaros, 1, 3] => Sulfuras, Hand of Ragnaros,1,3 
[Sulfuras, Hand of Ragnaros, 1, 10] => Sulfuras, Hand of Ragnaros,1,10 
[Sulfuras, Hand of Ragnaros, 1, 20] => Sulfuras, Hand of Ragnaros,1,20 
[Sulfuras, Hand of Ragnaros, 2, 0] => Sulfuras, Hand of Ragnaros,2,0 
[Sulfuras, Hand of Ragnaros, 2, 1] => Sulfuras, Hand of Ragnaros,2,1 
[Sulfuras, Hand of Ragnaros, 2, 2] => Sulfuras, Hand of Ragnaros,2,2 
[Sulfuras, Hand of Ragnaros, 2, 3] => Sulfuras, Hand of Ragnaros,2,3 
[Sulfuras, Hand of Ragnaros, 2, 10] => Sulfuras, Hand of Ragnaros,2,10 
[Sulfuras, Hand of Ragnaros, 2, 20] => Sulfuras, Hand of Ragnaros,2,20 
[Sulfuras, Hand of Ragnaros, 3, 0] => Sulfuras, Hand of Ragnaros,3,0 
[Sulfuras, Hand of Ragnaros, 3, 1] => Sulfuras, Hand of Ragnaros,3,1 
[Sulfuras, Hand of Ragnaros, 3, 2] => Sulfuras, Hand of Ragnaros,3,2 
[Sulfuras, Hand of Ragnaros, 3, 3] => Sulfuras, Hand of Ragnaros,3,3 
[Sulfuras, Hand of Ragnaros, 3, 10] => Sulfuras, Hand of Ragnaros,3,10 
[Sulfuras, Hand of Ragnaros, 3, 20] => Sulfuras, Hand of Ragnaros,3,20 
[Sulfuras, Hand of Ragnaros, 4, 0] => Sulfuras, Hand of Ragnaros,4,0 
[Sulfuras, Hand of Ragnaros, 4, 1] => Sulfuras, Hand of Ragnaros,4,1 
[Sulfuras, Hand of Ragnaros, 4, 2] => Sulfuras, Hand of Ragnaros,4,2 
[Sulfuras, Hand of Ragnaros, 4, 3] => Sulfuras, Hand of Ragnaros,4,3 
[Sulfuras, Hand of Ragnaros, 4, 10] => Sulfuras, Hand of Ragnaros,4,10 
[Sulfuras, Hand of Ragnaros, 4, 20] => Sulfuras, Hand of Ragnaros,4,20 
[Sulfuras, Hand of Ragnaros, 10, 0] => Sulfuras, Hand of Ragnaros,10,0 
[Sulfuras, Hand of Ragnaros, 10, 1] => Sulfuras, Hand of Ragnaros,10,1 
[Sulfuras, Hand of Ragnaros, 10, 2] => Sulfuras, Hand of Ragnaros,10,2 
[Sulfuras, Hand of Ragnaros, 10, 3] => Sulfuras, Hand of Ragnaros,10,3 
[Sulfuras, Hand of Ragnaros, 10, 10] => Sulfuras, Hand of Ragnaros,10,10 
[Sulfuras, Hand of Ragnaros, 10, 20] => Sulfuras, Hand of Ragnaros,10,20 
